### PR TITLE
[SC-185] Don't ever delete VMs from SRC cloud

### PR DESCRIPTION
**Pull request by antonf**

### CODE REVIEW

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferrylib.os.actions.vm_snapshot
cloudferrylib/os/actions/vm_snapshot.py:46: [W0613(unused-argument), VmSnapshot.run] Unused argument 'args'
cloudferrylib/os/actions/vm_snapshot.py:72: [W0613(unused-argument), CheckPointVm.run] Unused argument 'args'
cloudferrylib/os/actions/vm_snapshot.py:116: [W1202(logging-format-interpolation), VmRestore.run] Use % formatting in logging functions and pass the % parameters as arguments
cloudferrylib/os/actions/vm_snapshot.py:121: [W1202(logging-format-interpolation), VmRestore.run] Use % formatting in logging functions and pass the % parameters as arguments
cloudferrylib/os/actions/vm_snapshot.py:91: [W0613(unused-argument), VmRestore.run] Unused argument 'args'

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

File: cloudferrylib/os/actions/vm_snapshot.py [hunk 1]
```diff
@@ -92,36 +92,38 @@ def run(self, rollback_vars=None, *args, **kwargs):
...
                 continue
             if vm.id not in snapshot_from_namespace:
-                # delete this vm - we don't have its id in snapshot data
-                LOG.debug("vm {vm} will be deleted on {location}".format(
-                    vm=vm.id,
-                    location=self.cloud.position))
-                compute.delete_vm_by_id(vm.id)
+                if position == 'dst':
+                    # delete this vm - we don't have its id in snapshot data
+                    LOG.debug("vm {vm} will be deleted on {location}".format(
```
Comment (SVilgelm) [inline cloudferrylib/os/actions/vm_snapshot.py]: Rewrite this to use %(...)s formatting

File: cloudferrylib/os/actions/vm_snapshot.py [hunk 2]
```diff
@@ -92,36 +92,38 @@ def run(self, rollback_vars=None, *args, **kwargs):
...
+                        vm=vm.id,
+                        location=position))
+                    compute.delete_vm_by_id(vm.id)
             elif vm.status != snapshot_from_namespace.get(vm.id):
                 LOG.debug("status of {vm} is changed from {original}"
```
Comment (SVilgelm) [inline cloudferrylib/os/actions/vm_snapshot.py]: ditto

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferrylib.os.actions.vm_snapshot
cloudferrylib/os/actions/vm_snapshot.py:46: [W0613(unused-argument), VmSnapshot.run] Unused argument 'args'
cloudferrylib/os/actions/vm_snapshot.py:72: [W0613(unused-argument), CheckPointVm.run] Unused argument 'args'
cloudferrylib/os/actions/vm_snapshot.py:91: [W0613(unused-argument), VmRestore.run] Unused argument 'args'

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by roman-verchikov** [comment]
+1

**Comment by antonf** [comment]
recheck

**Comment by SVilgelm** [comment]
retest this please

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferrylib.os.actions.vm_snapshot
cloudferrylib/os/actions/vm_snapshot.py:46: [W0613(unused-argument), VmSnapshot.run] Unused argument 'args'
cloudferrylib/os/actions/vm_snapshot.py:72: [W0613(unused-argument), CheckPointVm.run] Unused argument 'args'
cloudferrylib/os/actions/vm_snapshot.py:91: [W0613(unused-argument), VmRestore.run] Unused argument 'args'

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by antonf** [comment]
retest this please

**Comment by antonf** [comment]
retest this please

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferrylib.os.actions.vm_snapshot
cloudferrylib/os/actions/vm_snapshot.py:46: [W0613(unused-argument), VmSnapshot.run] Unused argument 'args'
cloudferrylib/os/actions/vm_snapshot.py:72: [W0613(unused-argument), CheckPointVm.run] Unused argument 'args'
cloudferrylib/os/actions/vm_snapshot.py:91: [W0613(unused-argument), VmRestore.run] Unused argument 'args'

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by archyufa** [comment]
+1 tested in staging

**Pull request merged**